### ISSUE TITLE
Discrepancy between paper results and my replication

### ISSUE THREAD

**Issue by StevenMa616777** [wenxi-yue/SurgicalSAM#2]
Description:

Hello Wenxi,

I've been working with the code provided in this repository, trying to reproduce the results mentioned in your paper titled "SurgicalSAM: Efficient Class Promptable Surgical Instrument Segmentation". However, I observed a slight difference in the results.

Details of the discrepancy:

Expected Result (as per the paper):  mIoU(challenge IoU):0.8033 | IoU:0.8033 | mean class IoU:0.59 | IoU of each categories:
BF:0.84 | PF:0.66 | LND:0.59 | MCS:0.54 | UP:0.4 | SI:0.89 | CA:0.22

Observed Result: mIoU(challenge IoU):0.77868 | IoU:0.77868 | mean class IoU: 0.52833 | IoU of each categories:
BF:0.80748 | PF:0.62249 | LND:0.43873 | MCS:0.87306 | UP:0.21224 | SI:0.54476 | CA:0.19955

Steps to reproduce:
1.Cloned the repo
2.Installed all dependencies as mentioned
3.Download the checkpoints and data and place them in a right organisation
4.Ran the script inference.py with default settings

Environment:
OS: Ubuntu 20.04.5
Python version: 3.8.10
CUDA version: 11.7 
Pytorch version: torch 2.0.0+cu117, torchaudio 2.0.1+cu117, torchvision 0.15.1+cu117

I would appreciate any guidance or clarification on this. Thank you for your time!

**Comment by wenxi-yue** [comment]
Hi,

Thank you for your interest in our work.

We have tested the code from scratch again following the steps we provided, including cloning the repository, creating a new environment based on the provided instructions, and downloading the data and checkpoints from the link we shared. We successfully reproduced the results reported in the paper, as shown by the below screenshot on our server.

![ae3226239683b399dd45a892d766f61](https://github.com/wenxi-yue/SurgicalSAM/assets/142413487/522f48d1-d19e-494e-8bcb-c69d2a995acc)

We think that the performance discrepancy may be due to version differences of certain packages. Therefore, we provide more details about our system and conda environment below:

Operating System: Ubuntu 20.04.3
Conda Environment Packages and Versions: 
![1f38098038ddaa1fbe53e8d743aa2c7](https://github.com/wenxi-yue/SurgicalSAM/assets/142413487/fd40eef0-c8b9-48ca-aee9-2a62efc65462)
